lib/ansible/modules/cloud/amazon/iam_saml_federation.py

```diff
@@ -0,0 +1,191 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: iam_saml_federation
+version_added: "2.3"
+short_description: maintain iam saml federation configuration.
+requires: [ boto3 ]
+description:
+    - Maintains IAM SAML identity federation configurations.
+options:
+    name:
+        description:
+            - The name of the provider to create.
+        required: true
+    saml_metadata_document:
+        description:
+            - The XML document generated by an identity provider (IdP) that supports SAML 2.0.  More information on what this document is and how it is created is available in the AWS docs (http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_saml.html#idp-manage-identityprovider-console)
+    state:
+        description:
+            - Whether to create or delete identity provider. If 'present' is specified it will attempt to update the identity provider matching the name field.
+        default: present
+        choices: [ "present", "absent" ]
+extends_documentation_fragment:
+    - aws
+author: "Aidan Rowe (@aidan-)"
+'''
+
+EXAMPLES = '''
+# Note: None of these examples set aws_access_key or aws_secret_key.
+# It is assumed that their matching environment variables are set.
+
+# Creates a new iam saml identity provider if not present
+- name: saml provider
+    iam_saml_federation:
+        name: example1
+        saml_metadata_document: > # the > here opens an indented block, so no escaping/quoting is needed when in the indentation level under this key
+            <?xml version="1.0"?>...
+            <md:EntityDescriptor
+
+# Creates a new iam saml identity provider if not present
+- name: saml provider
+    iam_saml_federation:
+        name: example2
+        saml_metadata_document: "{{ item }}"
+    with_file: /path/to/idp/metdata.xml
+
+# Removes iam saml identity provider
+- name: remove saml provider
+    iam_saml_federation:
+        name: example3
+        state: absent
+'''
+
+RETURN = '''
+saml_provider_arn:
+    description: The ARN of the SAML Identity Provider that was created/modified.
+    type: string
+    sample: "arn:aws:iam::123456789012:saml-provider/my_saml_provider"
+'''
+
+try:
+    import boto3
+    import botocore.exceptions
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+import sys
+
+class SAMLProviderManager:
+    """Handles SAML Identity Provider configuration"""
+
+    def __init__(self, module, **aws_connection_params):
+        self.module = module
+        self.aws_connection_params = aws_connection_params
+
+        try:
+            self.conn= boto3_conn(module, conn_type='client', resource='iam', **self.aws_connection_params)
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(msg=str(e))
+
+    def _get_provider_arn(self, name):
+        providers = self.conn.list_saml_providers()
```

Comment:
These calls need to be wrapped in try/except to handle exceptions, otherwise the exception will just bubble up to the user. We provide a decorator in the EC2 module utils that handles boto3 exceptions and does retries for you. https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ec2.py#L71